x-pack/legacy/plugins/canvas/canvas_plugin_src/functions/browser/markdown.ts

```diff
@@ -4,10 +4,9 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { ExpressionFunction } from 'src/legacy/core_plugins/interpreter/public';
+import { Datatable, Render, Style, ExpressionFunction } from 'src/plugins/expressions/common';
```

Comment:
In our canvas/types we export everything from `src/plugins/expressions/common`.  We should be consistent in either importing from there or from src/plugins/expressions/common.  

I can clean that up later if you make an issue.

Reply:
I believe we should be importing/exporting from `src/plugins/expressions/public` instead.

`common` is not a concept in the new platform like it is in the legacy world, just an implementation detail of the plugin itself, and therefore isn't considered part of the "official" API for a service.

Reply:
Canvas server-side was importing client-side code of legacy Interpreter.

![image](https://user-images.githubusercontent.com/9773803/68966685-f99a6200-07de-11ea-8ddb-b77f1184b351.png)

It somehow worked in the legacy platform. But in the New Platform it just crashed without any clear error message, I had to do quite some debugging to find the reason. That's why I went through all those Canvas imports to change them to something that works. Currently Canvas has to import from `expressions/common` as we don't have server-side in Expressions service from where Canvas could import. I've [created an issue](https://github.com/elastic/kibana/issues/50808).

Reply:
We could re-export everything from the server-side, but given that we don't have server-side now, just blindly re-exporting everything maybe is not what we want. Also, you need to know which Canvas function/type is client-side and which one is server-side (or maybe both?), which I currently don't really want to investigate.

Also, the `x-pack/legacy/plugins/canvas/types/index.ts` file, mentioned by @crob611, where canvas re-exports Expression types&mdash;I believe it is used by, both, client- and server-side. Thus, I had to change that file a bit to only import from `/common`.

Also, I'm not sure if we should ban imports from `/common` folder. It seems like it could be a good place from which server- and client-side could import; without needing to re-export from `/public` and `/server`.